integration/spark/src/main/scala/org/apache/carbondata/spark/util/CommonUtil.scala

```diff
@@ -1011,38 +1011,46 @@ object CommonUtil {
       objectDataType: DataType): AnyRef = {
     objectDataType match {
       case _: ArrayType =>
-        val arrayDataType = objectDataType.asInstanceOf[ArrayType]
-        val arrayData = data.asInstanceOf[UnsafeArrayData]
-        val size = arrayData.numElements()
-        val childDataType = arrayDataType.elementType
-        val arrayChildObjects = new Array[AnyRef](size)
-        var i = 0
-        while (i < size) {
-          arrayChildObjects(i) = convertSparkComplexTypeToCarbonObject(arrayData.get(i,
-            childDataType), childDataType)
-          i = i + 1
+        if (data == null) {
```

Comment:
after line 1011, add a check that if data type is array or struct or map and the data is null, return null to avoid changing many lines.

Reply:
For dateType if data is null that is handled by some other way which is not returning null. That's why i didn't put it after line 1011

Reply:
For dateType if data is null that is handled by some other way which is not returning null. That's why i didn't put it after line 1011

Reply:
Done